opal/corelib/struct.rb

```diff
@@ -83,7 +83,15 @@ def [](name)
 
       name = members[name]
     elsif String === name
-      raise NameError.new("no member '#{name}' in struct", name) unless members.include?(name.to_sym)
+      %x{
+        if(!self.$$data.hasOwnProperty(name)) {
+          Opal.Kernel.$raise(
+            Opal.NameError.$new(
+              "no member '" + name + "' in struct", name
+            )
```

Comment:
I had some trouble with the interpolation. I can't remember what it was, but it was really late last night and I don't think I paid attention to the failure so I just ended up doing the thing I knew would work. Maybe doing it when I'm more awake would help. :-)